lib/teleterm/services/connectmycomputer/connectmycomputer_test.go

```diff
@@ -88,27 +88,34 @@ func TestRoleSetupRun_Idempotency(t *testing.T) {
 	_, err = roleSetup.Run(ctx, accessAndIdentity, certManager, &clusters.Cluster{URI: uri.NewClusterURI("foo")})
 	require.NoError(t, err)
 
-	require.Equal(t, 1, accessAndIdentity.callCounts["UpsertRole"], "expected two runs to update the role only once")
+	require.Equal(t, 1, accessAndIdentity.callCounts["CreateRole"], "expected two runs to create the role only once")
+	require.Equal(t, 0, accessAndIdentity.callCounts["UpdateRole"], "expected two runs to not update the role")
 	require.Equal(t, 1, accessAndIdentity.callCounts["UpdateUser"], "expected two runs to update the user only once")
 }
 
 func TestRoleSetupRun_RoleErrors(t *testing.T) {
 	existingRole, err := types.NewRole("connect-my-computer-alice", types.RoleSpecV6{})
 	require.NoError(t, err)
 
+	bogusErr := errors.New("something went wrong")
+
 	tests := []struct {
 		name          string
-		upsertRoleErr error
 		existingRole  types.Role
+		createRoleErr error
+		updateRoleErr error
+		wantErr       error
 	}{
 		{
 			name:          "creating role fails",
-			upsertRoleErr: errors.New("something went wrong"),
+			createRoleErr: bogusErr,
+			wantErr:       bogusErr,
```

Comment:
`wantErr` is always equal to `bogusErr`, but I decided to pass it explicitly to make the tests easier to read. Please let me know if there's a more idiomatic way to handle tests like these.

Reply:
Seems alright to me.

I suppose you could do something like `wantErr := bogusErr` outside of the test table, but your approach is more resilient to future changes.